threading/channels.nim

```diff
@@ -187,8 +187,9 @@ proc channelSend(chan: ChannelRaw, data: pointer, size: int, blocking: static bo
 
   when not blocking:
     if chan.isFull(): return false
-
-  acquire(chan.lock)
+    if not tryAcquire(chan.lock): return false
```

Comment:
Sorry, the documentation states that this version of send doesn't block. I assume it would need to use a non-blocking version for acquiring the locks. I made two multiple producers-multiple readers benchmarks and I see high contention. Notice that there's a single lock used by both producers and consumers and I assume that's the primary reason for the high contention. Using `acquire` instead creates slightly better numbers but I wonder if that's just a side-effect from the fairness guarantees that are implemented for normal locks. Then I implemented exponential backoff for both benchmarks. The success rates improved to 98%, it's clearly the better mechanism to fix the issue. As such I claim that `acquire` is not needed here, if anything trySend/tryRecv now behaves similarly to same named functions from a lock-less channel.